### PR TITLE
widalarmeta: Selectable font. Optional hour padding.

### PR DESCRIPTION
**Pull request by c-mauderer**
Allow to select between bitmap and teletext font. Allow to disable the hour padding.

### CODE REVIEW

File: apps/widalarmeta/widget.js [hunk 1]
```diff
@@ -49,17 +52,34 @@
...
+      var extraPixels = 2;
+      var nrDigits = 5;
+
+      var text = "";
+      if (config.padHours) {
+        text += hours.padStart(2, '0');
+      } else {
+        text += hours;
+        if (hours < 10) {
+          nrDigits -= 1;
```
Comment (nxdefiant) [inline apps/widalarmeta/widget.js:65]: Wouldn't it be easier to calculate the width
from the string length?

Comment (c-mauderer) [reply apps/widalarmeta/widget.js:65]: You are most likely right. I clearly haven't thought enough about it. It was a fixed value before and I just adapted it. I'll change that.

Comment (c-mauderer) [reply apps/widalarmeta/widget.js]: @nxdefiant: I used the `g.stringWidth(...)` function instead. With that it is even easier to add further fonts. As an example I added another 4x5 font option.

Comment (nxdefiant) [reply apps/widalarmeta/widget.js]: cool and a lot easier to read

**Comment by gfwilliams** [comment]
Thanks! merging now.

**Comment by c-mauderer** [comment]
> Thanks! merging now.

Thank you.

**Pull request merged**